frb_example/pure_dart/rust/src/api.rs

```diff
@@ -1156,6 +1156,13 @@ pub fn run_opaque(opaque: RustOpaque<HideData>) -> String {
     opaque.hide_data()
 }
 
+#[allow(clippy::redundant_clone)]
+pub fn run_opaque_inner(opaque: RustOpaque<HideData>) -> String {
+    // Tests whether `.clone()` works even without the generic type wrapped by it
+    // implementing Clone.
+    opaque.clone().hide_data()
```

Comment:
uber-nit: To make it explicit and avoid future programmers accidentically adding Clone to HideData, what about this: We can create a *new* struct with *explicit* name saying it is not clone. e.g. `struct NonCloneableData { a: String }`, and use it here

Reply:
Well, I made sure to check that HideData isn't Clone.

Reply:
How did you do that? If it is done automatically then also OK.

This is because, we have to imagine that a future programmer may do some refactoring or adding features/tests that may break current code if there is no *automated* tests

Reply:
I just did it manually. You're right, if it ever got added, the whole test wouldn't work any more (well, it wouldn't fail, but it'd work with the old code as well).

Reply:
Yes that's the problem - then the test will not test what you want to test anymore